deployer/deployer_test.go

```diff
@@ -5,71 +5,61 @@ import (
 	"context"
 	"encoding/hex"
 	"encoding/json"
+	"log"
 	"os"
-
 	"testing"
 
 	"github.com/golang/mock/gomock"
+	"github.com/joho/godotenv"
 	"github.com/pkg/errors"
 	"github.com/stretchr/testify/assert"
-
 	"github.com/threefoldtech/grid3-go/mocks"
 	client "github.com/threefoldtech/grid3-go/node"
 	"github.com/threefoldtech/grid3-go/subi"
 	"github.com/threefoldtech/grid3-go/workloads"
-
 	"github.com/threefoldtech/substrate-client"
-
 	"github.com/threefoldtech/zos/pkg/gridtypes"
 	"github.com/threefoldtech/zos/pkg/gridtypes/zos"
-
-	"github.com/joho/godotenv"
 )
 
-var (
-	SubstrateURLs = map[string]string{
-		"dev":  "wss://tfchain.dev.grid.tf/ws",
-		"test": "wss://tfchain.test.grid.tf/ws",
-		"qa":   "wss://tfchain.qa.grid.tf/ws",
-		"main": "wss://tfchain.grid.tf/ws",
-	}
-)
+var backendURLWithTLSPassthrough = "//1.1.1.1:10"
+var backendURLWithoutTLSPassthrough = "http://1.1.1.1:10"
 
-func SetUP() (identity subi.Identity, twinID uint32, err error) {
-	if _, err = os.Stat("../.env"); !errors.Is(err, os.ErrNotExist) {
-		err = godotenv.Load("../.env")
+func setup() (TFPluginClient, error) {
+	if _, err := os.Stat("../.env"); !errors.Is(err, os.ErrNotExist) {
+		err := godotenv.Load("../.env")
 		if err != nil {
-			return
+			return TFPluginClient{}, err
 		}
 	}
 
 	mnemonics := os.Getenv("MNEMONICS")
-	identity, err = substrate.NewIdentityFromSr25519Phrase(mnemonics)
-	if err != nil {
-		return
-	}
-
-	keyPair, err := identity.KeyPair()
-	if err != nil {
-		return
-	}
+	log.Printf("mnemonics: %s", mnemonics)
 
 	network := os.Getenv("NETWORK")
-	pub := keyPair.Public()
-	sub := subi.NewManager(SubstrateURLs[network])
-	subext, err := sub.SubstrateExt()
-	if err != nil {
-		return
-	}
-	twin, err := subext.GetTwinByPubKey(pub)
+	log.Printf("network: %s", network)
+
+	return NewTFPluginClient(mnemonics, "sr25519", network, "", "", true, "", true)
```

Comment:
for log list of arguments please use overlay e.g `NewTFPluginClient(mnemonics /*mnemonics*/, "sr25519" /*key type*/ ... etc)`